scm-ui/ui-components/src/devBuild.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020-present Cloudogu GmbH and Contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+export const isDevBuild = () => (process.env.NODE_ENV === "Development")
+
+export const appendTestId = (testId?: string) => isDevBuild() ? testId : undefined;
```

Comment:
I think something like the following would be nicer:

```suggestion
export const createAttributesForTesting = (testId?: string) => {
  if (!testId || isDevBuild()) {
    return; // not sure if undefined ok, perhaps we have to return an empty object
  }
  return {
    "data-testid": testId
  }
};
```

This could be used as follows:

```javascript
<button {...createAttributesForTesting("mytestid")} ... />
```

With this way, we are open for other test attributes if we need them and we do not render anything in production.